### PR TITLE
Allow LMDB to be opened by multiple readers simultaneously

### PR DESCRIPTION
**Pull request by bowang**
This PR is basically a one-line change by adding MDB_NOLOCK flag. Since LMDBReader is read-only, we don't need locks. LMDB by default creates a lock file in the same folder which prohibits the database to be opened by others.

### CODE REVIEW

**Comment by sb2nov** [comment]
Jenkins, test this please

**Comment by bowang** [comment]
I have some difficulties at understanding the failures.
* Linux GPU: failed at several tests irrelevant to what's being modified.
//tensorflow/contrib/signal:window_ops_test
//tensorflow/examples/adding_an_op:cuda_op_test
//tensorflow/python:adam_test
//tensorflow/python/kernel_tests:norm_op_test
* Linux CPU Tests: can someone help me get the log?
//tensorflow/python/kernel_tests:reader_ops_test
`/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/org_tensorflow/bazel-out/local-opt/testlogs/tensorflow/python/kernel_tests/reader_ops_test/test.log`

File: tensorflow/python/kernel_tests/reader_ops_test.py [hunk 1]
```diff
@@ -1011,6 +1014,25 @@ def testReadFromFile(self):
...
+      reader1 = io_ops.LMDBReader(name="test_read_from_same_file1")
+      reader2 = io_ops.LMDBReader(name="test_read_from_same_file2")
+      filename_queue = input_lib.string_input_producer([self.db_path],
+                                                       num_epochs=None)
+      key1, value1 = reader1.read(filename_queue)
+      key2, value2 = reader2.read(filename_queue)
+
+      coord = coordinator.Coordinator()
+      threads = queue_runner_impl.start_queue_runners(sess, coord=coord)
+      for i in range(0, 3):
```
Comment (jhseu) [inline tensorflow/python/kernel_tests/reader_ops_test.py]: range(3)

File: tensorflow/python/kernel_tests/reader_ops_test.py [hunk 2]
```diff
@@ -1011,6 +1014,25 @@ def testReadFromFile(self):
...
+        for j in range(0, 10):
```
Comment (jhseu) [inline tensorflow/python/kernel_tests/reader_ops_test.py]: range(10)

**Review by jhseu — APPROVED** [review]

File: tensorflow/python/kernel_tests/reader_ops_test.py (hunk 1 above)
Comment (bowang) [reply tensorflow/python/kernel_tests/reader_ops_test.py]: Fixed. Thanks!

File: tensorflow/python/kernel_tests/reader_ops_test.py (hunk 2 above)
Comment (bowang) [reply tensorflow/python/kernel_tests/reader_ops_test.py]: Fixed. Thanks!

**Comment by bowang** [comment]
@jhseu is there anything I can do to get it merged?

**Comment by bowang** [comment]
@caisq can you kick off the Jenkins test?

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by bowang** [comment]
@caisq Can we kick off another jenkins test?
I just merged the latest master. Hope the failures are resolved in master branch already.

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by gunan** [comment]
Jenkins, test this please.

**Comment by snnn** [comment]
Hi,

The function mdb_env_setup_locks in mdb.c is not thread-safe. 
```c
		if (!mdb_sec_inited) {
			InitializeSecurityDescriptor(&mdb_null_sd,
				SECURITY_DESCRIPTOR_REVISION);
			SetSecurityDescriptorDacl(&mdb_null_sd, TRUE, 0, FALSE);
			mdb_all_sa.nLength = sizeof(SECURITY_ATTRIBUTES);
			mdb_all_sa.bInheritHandle = FALSE;
			mdb_all_sa.lpSecurityDescriptor = &mdb_null_sd;
			mdb_sec_inited = 1;
		}
```
mdb_sec_inited is a global var.


I'm not sure if this function will be called, but, let's double-check it.

**Pull request merged**